### ISSUE TITLE
Vector3.PointOnLine returns bad results for long lines. 

### ISSUE THREAD

**Issue by DmytroMuravskyi** [hypar-io/Elements#832]
Vector3.PointOnLine is based on dot product and because of that answer is angle based. The longer the line is - the bigger deviation is allowed. For example Line((0, 0, 0), (150, 0, 0) will return true for (100, 0.1, 0) point that can lead to unwanted results.
Because how `includeEnds` parameter is coded:
```
        public static bool PointOnLine(Vector3 point, Vector3 start, Vector3 end, bool includeEnds = false)
        {
            if (includeEnds && (point.DistanceTo(start) < Vector3.EPSILON || point.DistanceTo(end) < Vector3.EPSILON))
            {
                return true;
            }
            return (start - point).Unitized().Dot((end - point).Unitized()) < (Vector3.EPSILON - 1);
        }
```
it's unclear what it means: when it's false exactly end point will always return true, but when numeric noise is present:
(1e-6, 0, 0) will return True for the line above but (1e-6, 1e-6, 0) will return false. 
This contradicts with IsAlmostEqualTo.

In my opinion, PointOnLine should be distance based and be consistent with Vector3.DistanceTo(Line): return true if point is less than tolerance away from the line. 
Only alternative solution I see is to multiply tolerance based on line length but it feels really bad.
My proposal is next:
```       
        public static bool PointOnLine(Vector3 point, Vector3 start, Vector3 end, bool includeEnds = false)
        {
            if (includeEnds && (point.IsAlmostEqualTo(start) || point.IsAlmostEqualTo(end)))
            {
                return true;
            }

            var delta = end - start;
            var lambda = (point - start).Dot(delta) / (end - start).Dot(delta);
            if( lambda > 0 && lambda < 1)
            {
                var pointOnLine = start + lambda * delta;
                return pointOnLine.IsAlmostEqualTo(point);
            }
            return false;
        }
```
IsAlmostEqualTo is not using Sqrt so it's better for checks than DistanceTo. lambda calculation is taken from Vector3.DistanceTo(Line).
This provides similar behavior for end points: when includeEnds is false the exact same point will return false but both (1e-6, 0, 0) and (1e-6, 1e-6, 0) will not return true. If we want the version where lf  includeEnds  is false then we return false for any points in proximity of end points we can do something like this:
```       
        public static bool PointOnLine(Vector3 point, Vector3 start, Vector3 end, bool includeEnds = false)
        {
            if (!includeEnds && (point.IsAlmostEqualTo(start) || point.IsAlmostEqualTo(end)))
            {
                return false;
            }

            var delta = end - start;
            var lambda = (point - start).Dot(delta) / (end - start).Dot(delta);
            if( lambda > 0 && lambda < 1)
            {
                var pointOnLine = start + lambda * delta;
                return pointOnLine.IsAlmostEqualTo(point);
            }
            return false;
        }
```

**Comment by andrewheumann** [comment]
I like the first proposal — feel free to send me a PR. As long as we're clear in the documentation comments about what `includeEnds` does, then I think it's fine.

**Comment by andrewheumann** [comment]
This is resolved with #836
